codalab/rest/account.py

```diff
@@ -101,7 +108,9 @@ def do_signup():
         })
```

Comment:
You need to provide the new fields to this redirect call as well. This allows the sign up page to display the already-filled values again when it displays an error. (If you test it, you will see that only email and username currently gets filled out again after an error, such as mismatched passwords, occurs.)